nsot/models.py

```diff
@@ -1396,8 +1396,43 @@ def set_addresses(self, addresses, overwrite=False, partial=False):
 
         self.clean_addresses()
 
+    def get_ancestors(self):
+        """Recursively get all parents of an interface"""
+        p = self.parent
+        ancestors = []
```

Comment:
Just a comment: Returning a list instead of a queryset may be problematic if we attempt to implement any filtering (or filter chaining) for ancestors. This isn't currently a use-case but it will be something to keep in mind if the need arises in the future.